### PR TITLE
Updating exploring.md

### PR DESCRIPTION
**Pull request by allanaaa**
Images still need some trimming. Haven't explained logs yet.

### CODE REVIEW

**Comment by wetneb** [comment]
By "logs" do you mean OpenRefine server logs? I am not sure I would describe those there. It would be great to document how to access them, but probably in Running or some troubleshooting section instead?

**Comment by allanaaa** [comment]
One thing I'd like feedback on:
The TOC sidebar is pretty long, and the facets section could use some breaking-up of its content. As you know, we only get two levels of hierarchy in this menu, so maybe we should move some things out of "Facets" and into their own grouping: "_____ facets" and "Customized facets", e.g. I'm not sure what to name this category. Default facets? Simple facets? Basic facets? Original facets?  

I do want all the facets currently listed to appear, but I can entertain arguments for demoting some of them (e.g. just showing "Customized facets" and then demoting everything under that to an h4 and making them disappear from the TOC). 

sidebar

**Comment by thadguidry** [comment]
@allanaaa Another option is to actually create a GeneralFacets.md file as well as a CustomizedFacets.md file?  Then provide links to them in the exploring.md file?

**Comment by allanaaa** [comment]
Oh yes, thanks! We had discussed splitting this page in two but I forgot to mention it here. 
I am not sure the precise structure you mention will work - I think it would make more sense to have "Exploring data" in the left-hand menu be a category containing multiple pages. Splitting this page into three pages as
- cell data types and record types
- facets and filter
- sort and view

would make the most sense to me, rather than having one page with two more sub-pages tucked inside the facets section. But I can experiment with that and see it's more seamless than I imagine, if there is interest.

**Comment by thadguidry** [comment]
@allanaaa Yes, something like that... sure, let's try your suggestion to see how that feels!

File: docs/docs/manual/exploring.md
Comment (wetneb) [inline docs/docs/manual/exploring.md:56]: It's not clear to me that this is the single most salient sort of issue we should flag about the records mode - perhaps it could be left in normal text without the caution box (which really jumps to the eyes).

File: docs/docs/manual/exploring.md
Comment (wetneb) [inline docs/docs/manual/exploring.md:135]: The editing part might be worth expanding, it's a powerful feature that is not so easy to discover. It would be worth explaining that this will change all the corresponding cell values: it is an effective way to clean up values in a column. You could add a link to the documentation about clustering (to be written) which does that at a larger scale.

Comment (thadguidry) [reply docs/docs/manual/exploring.md:135]: We internally call it a "mass edit" link, because it can massively edit from/to all the values in a column as Antonin says.

**Comment by allanaaa** [comment]
@thadguidry mentioned nanosecond support on date formats. I just tested it out and it doesn't seem to be working. 
Assuming this format is correct: 2014-03-14T05:30:04.000789000Z from https://github.com/OpenRefine/OpenRefine/wiki/GREL-Date-Functions

OpenRefine either 
- truncates the .nnnn value, which it did on a toDate column operation, and when I changed the cell data type on "2014-03-14T05:30:04.000789000" (without the Z at the end)
- or it actually misconverts. It changed "2014-03-14T05:30:04.000789000Z" as text to "2014-03-14T01:30:04Z" as a date - pretty sure nanoseconds don't allow us the ability to go back in time, but I'm not a physicist. 

I assume we want neither truncation nor misconversion. 

Also, ISO 8601 states that it slightly prefers a comma instead of a decimal, but the toDate function doesn't support the comma at all (won't convert it). I think I should probably document that, yes?

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md]: I added a sentence or two. I can probably say more here so I'll leave this open for now. Current text: 

> You can edit any entry that appears in the facet display, by hovering over the facet and clicking the “edit” button that appears. This will mass-edit every matching cell in your data. This is a great way to fix typos, whitespace, and other issues that may be affecting the way facets appear. You can also automate the cleanup of facets by using [clustering](transforming#cluster-and-edit), with the “Cluster” button displayed within the facet window.

**Comment by allanaaa** [comment]
I've ripped most of the date-format information out of this page, since it's more about the transform operation & the GREL function. I've just expanded a bit on the cell data type with links out. We can deal with Date.parse and SimpleDateFormat later!
My comment above should probably be an issue on its own (assuming I'm not totally off-base with it). 

The last part of the cell-data-types section now reads 

> Converting a cell's data type is not the same operation as transforming its contents. For example, using a column-wide transform such as “Transform” > “Common transforms …” > “to date” may not convert all values successfully, but going to an individual cell, clicking “edit” and changing the data type can successfully convert text to a date.
> To transform data from one type to another, see [Transforming data](transforming#transform) for information on using common tranforms, and see [Expressions](expressions) for information on using `toString()`, `toDate()`, and other functions. 

... which I think is more or less correct based on our chat today, but is probably too vague to be helpful at this point in the documentation. I have written this as a "for example" but perhaps date formatting is the only case in which this is really valid (or important to understand). Thoughts?

**Comment by tfmorris** [comment]
> @thadguidry mentioned nanosecond support on date formats. I just tested it out and it doesn't seem to be working.
> Assuming this format is correct: 2014-03-14T05:30:04.000789000Z from https://github.com/OpenRefine/OpenRefine/wiki/GREL-Date-Functions

A bunch of the examples (e.g. datePart(millis)) on that page are wrong, so I'm not sure I'd trust it. 

> OpenRefine either
> 
> * truncates the .nnnn value, which it did on a toDate column operation, and when I changed the cell data type on "2014-03-14T05:30:04.000789000" (without the Z at the end)
> * or it actually misconverts. It changed "2014-03-14T05:30:04.000789000Z" as text to "2014-03-14T01:30:04Z" as a date - pretty sure nanoseconds don't allow us the ability to go back in time, but I'm not a physicist.
> 
> I assume we want neither truncation nor misconversion.

Correct. Please file bug reports, if you haven't already done so. I don't think nanoseconds are supported, but it should give a reasonable error message. What timezone are you in? The 1:30 vs 5:30 thing sounds suspiciously like a timezone bug in the new date code (which is incredibly messed up).

> Also, ISO 8601 states that it slightly prefers a comma instead of a decimal, but the toDate function doesn't support the comma at all (won't convert it). I think I should probably document that, yes?

Yes, please, but as this comment https://github.com/moment/moment/issues/2580#issuecomment-136226383 says, pretty much everywhere in computer land uses a period decimal separator.

I think I'm missing some nuance of the difference between "convert" and "transform," but the behavior sounds wrong from the description. For cases of wrong behavior, we should: 1) file a bug report, 2) document the current behavior, but with a link to the issue for the correct behavior so that people know when it gets fixed.

**Comment by thadguidry** [comment]
@tfmorris Nanosecond support landed here: PR #1597
Please also read #1287 and also maybe a bit of #1998 
I also went searching through my chat with Jacky at one point in time (which coincides with his [May 10, 2018 comment in #1597](https://github.com/OpenRefine/OpenRefine/pull/1597#issuecomment-388091137))...

> Qi (Jacky) Cui <cuiqi1202@gmail.com>
> Wed, May 9, 2018, 9:24 PM
> will add Nano to inc

**Comment by tfmorris** [comment]
> Nanosecond support landed here: PR #1597

That is a refactoring PR, which, by definition, shouldn't include any functional changes, but it apparently has a half-baked attempt at nanosecond support. What it didn't do, which should have been part of the refactoring, was switch from the use of `SimpleDateFormat` or `GregorianCalendar`. Since `SimpleDateFormat` is used by `ToDate()` and doesn't support nanosecond precision, it's the likely cause of at least some of the bugs that @allanaaa is running into.

**Comment by allanaaa** [comment]
Something I forgot to mention earlier:

Currently a section is titled "Row types: rows vs. records", which is unwieldy. I would love suggestions for a snappier title about records mode.

**Comment by ostephens** [comment]
>Row types: rows vs. records
Why not just "Rows vs. Records"?

File: docs/docs/manual/exploring.md
Comment (ostephens) [inline docs/docs/manual/exploring.md:52]: I'd suggest that an ID of some kind would be a better example of a key column - the scenario you describe would work in a sense I think a unique ID that links the rows together is a better example of how records work

**Comment by allanaaa** [comment]
Something I forgot to ask about, in the Sort menu:

> *   boolean values by false, then true, or by **core-views/true-false**

coreviews

I don't know what "core-views" means! Can anyone tell me?

**Comment by thadguidry** [comment]
@allanaaa That's a bug... it's a placeholder for translated text in javascript... but looks like someone put in the literal string itself "core-views/true-false" and so it is not getting translated.
File a bug for this please, Thanks!

**Comment by tfmorris** [comment]
> Currently a section is titled "Row types: rows vs. records", which is unwieldy. I would love suggestions for a snappier title about records mode.

Simply "Records vs Rows"? I find the "Row types:" prefix to be confusing. It's like saying "Apple types: Apples vs. Oranges". A record is made up of multiple rows.

Another possibility might be "Display modes" or something similar.

File: docs/docs/manual/facets.md
Comment (ostephens) [inline docs/docs/manual/facets.md:93]: I don't think this is correct. Blank, non-numeric and error values don't appear as zero values - they aren't included in the bar chart - they are included/excluded based on checking the relevant boxes

File: docs/docs/manual/facets.md
Comment (ostephens) [inline docs/docs/manual/facets.md:169]: I'm not sure I'd say "simple". It's a custom text facet

File: docs/docs/manual/facets.md
Comment (ostephens) [inline docs/docs/manual/facets.md:115]: Is it worth mentioning somewhere that (unlike a Text facet) a Custom text facet does not support editing facet values to update the data in the cells used to create the facet?

File: docs/docs/manual/facets.md
Comment (ostephens) [inline docs/docs/manual/facets.md:242]: I'd suggest "To store errors in cells" rather than "To output errors"

File: docs/docs/manual/facets.md
Comment (ostephens) [inline docs/docs/manual/facets.md:276]: Is it worth giving a simple text filter example as well as a regular expression version?

File: docs/docs/manual/facets.md
Comment (ostephens) [inline docs/docs/manual/facets.md:189]: I'm not sure "Logarithmic scales reduce wide-ranging quantities to tiny scopes" is very clear. What are "tiny scopes"?

I think the description from wikipedia is better:
"A logarithmic scale (or log scale) is a way of displaying numerical data over a very wide range of values in a compact way"

File: docs/docs/manual/sortview.md
Comment (ostephens) [inline docs/docs/manual/sortview.md:19]: I think an explanation of in what sense sorts are usually not "permanent" (i.e. the original row/record order is preserved which can be seen by the numbers in the All column) and what making it "permanent" means (rows/records are renumbered sequentially based on the current sort and the "sort" disappears

Also think we need an explanation of how multiple sorts apply - i.e. if you apply multiple sorts at once, they are ordered and don't override each other (sort by author then title preserves the author orders and sorts titles within a single author).

This links to making sorts "permanent" as to get a sort of books by author then by title within author you'd want to:

Apply an author sort
Apply a title sort
Make the sort permanent

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md:15]: On import, OpenRefine considers everything a string at the preview stage, right? That's my experience. Are there specific file formats that import differently? 

I know in preview you can explicitly tell it to parse numbers. But we're no longer in "default" by then.

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md:25]: Okay, we've removed arrays entirely from this section. 
It now reads:

> The data types supported are:
> 
>     string (one or more text characters)
>     number (one or more characters of numbers only)
>     boolean (values of “true” or “false”)
>     date (ISO-8601-compliant extended format with time in UTC: YYYY-MM-DDTHH:MM:SSZ)
> 
> A “date” type is created when a text column is transformed into dates, or when individual cells are set to have the data type “date.”
> 
> OpenRefine recognizes two further data types as a result of its own processes:
> 
>     error
>     null
> 
> An “error” data type is created when the cell is storing an error generated during a transformation in OpenRefine.
> 
> A “null” data type is a special value which basically means “this cell has no value.”...

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md]: I'm not sure why you want a unique ID for this. I think this example works well with just the film title, not the ID?

|IMDB ID|Work|Actor|Role|
|---|---|---|---|
|tt0032138|The Wizard of Oz|Judy Garland|Dorothy Gale|
|||Ray Bolger|"Hunk"|
||||The Scarecrow|
|||Jack Haley|"Hickory"|
||||The Tin Man|
|||Bert Lahr|"Zeke"|
||||The Cowardly Lion|
|||Frank Morgan|Professor Marvel|
||||The Gatekeeper|
||||The Carriage Driver|
||||The Guard|
||||The Wizard of Oz|
|||Margaret Hamilton|Miss Almira Gulch|
||||The Wicked Witch of the West|

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md]: I hope "at first" is an acceptable compromise here. I still think it's a default. :)

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md]: I've removed the unique ID for now - maybe the section needs more explication about how you don't actually need a key column of unique IDs (because OR does that work behind the scenes), but of course your data may need to be sorted (or un-sorted) not to confuse the issue. But I think that can wait for a second pass.

**Comment by allanaaa** [comment]
I'm ready to merge this PR if everyone else is!

File: docs/docs/manual/exploring.md
Comment (ostephens) [reply docs/docs/manual/exploring.md]: No I don't think that's true. Importers will have rules about what they do, and it's definitely possible to have an importer that defaults to all values being treated as strings but that's not a general rule. If you import from Excel behaviour is different to if you import from CSV. I agree that certain importers and the UI has some default behaviour here, but this isn't OpenRefine wide "default" - its specific to those scenarios.

**Comment by ostephens** [comment]
I can't agree with the "default" or "at first" phrasing with regarded to cell data types - I think we need to be clearer here. I've just responded to your comment.

My ideal is that we fix this before the merge, but will be OK to merge as long as there is an issue for this so we don't forget to revisit

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md]: I haven't seen any evidence of this behaviour but I'm happy to believe you if you say so. 

What about "Most of the time, every cell will be considered a string, but...." or something like that? It does seem to be the vast majority of cases.

Comment (tfmorris) [reply docs/docs/manual/exploring.md]: OpenRefine importers preserve the data types for any format that supports them - Excel (XLS & XLSX), OpenOffice, Google Sheets, JSON, XML, probably others...

**Comment by wetneb** [comment]
Let's fix this before merging then :)

File: docs/docs/manual/exploring.md
Comment (allanaaa) [reply docs/docs/manual/exploring.md]: > Each piece of information (each cell) in OpenRefine is assigned a data type. Some file formats, when imported, can set data types that are recognized by OpenRefine. Cells without an associated data type on import will be considered a “string” at first, but you can have OpenRefine convert cell contents into other data types later.

🤷‍♀️

**Comment by ostephens** [comment]
Happy with @allanaaa's latest suggestion

**Review by ostephens — APPROVED** [review]

**Pull request merged**
